src/codemagic/tools/app_store_connect/actions/fetch_signing_files_action.py

```diff
@@ -0,0 +1,237 @@
+        self,
+        bundle_id_identifier: str,
+        platform: BundleIdPlatform,
+        create_resource: bool,
+        strict_match: bool,
+    ) -> List[BundleId]:
+        bundle_ids = self.list_bundle_ids(
+            bundle_id_identifier,
+            platform=platform,
+            bundle_id_identifier_strict_match=strict_match,
+            should_print=False,
+        )
+        if not bundle_ids:
+            if not create_resource:
+                raise AppStoreConnectError(f"Did not find {BundleId.s} with identifier {bundle_id_identifier}")
+            bundle_ids.append(self.create_bundle_id(bundle_id_identifier, platform=platform, should_print=False))
+        else:
+            for bundle_id in bundle_ids:
+                self.logger.info(f"- {bundle_id.attributes.name} {bundle_id.attributes.identifier} ({bundle_id.id})")
+        return bundle_ids
+
+    def _get_or_create_certificates(
+        self,
+        profile_type: ProfileType,
+        certificate_key: Optional[Union[PrivateKey, Types.CertificateKeyArgument]],
+        certificate_key_password: Optional[Types.CertificateKeyPasswordArgument],
+        create_resource: bool,
+    ) -> List[SigningCertificate]:
+        certificate_types = [CertificateType.from_profile_type(profile_type)]
+        # Include iOS and Mac App distribution certificate types backwards compatibility.
+        # In the past iOS and Mac App Store profiles used to map to iOS and Mac App distribution
+        # certificates, and we want to keep using existing certificates for as long as possible.
+        if profile_type is ProfileType.IOS_APP_STORE:
+            certificate_types.append(CertificateType.IOS_DISTRIBUTION)
+        elif profile_type is ProfileType.IOS_APP_ADHOC:
+            certificate_types.append(CertificateType.IOS_DISTRIBUTION)
+        elif profile_type is ProfileType.MAC_APP_STORE:
+            certificate_types.append(CertificateType.MAC_APP_DISTRIBUTION)
+        elif profile_type is ProfileType.MAC_APP_DIRECT:
+            certificate_types.append(CertificateType.DEVELOPER_ID_APPLICATION)
+
+        certificates = self.list_certificates(
+            certificate_types=certificate_types,
+            certificate_key=certificate_key,
+            certificate_key_password=certificate_key_password,
+            should_print=False,
+        )
+
+        if not certificates:
+            if not create_resource:
+                _certificate_types = ", ".join(map(str, certificate_types))
+                raise AppStoreConnectError(f"Did not find {SigningCertificate.s} with type {_certificate_types}")
+            certificate = self.create_certificate(
+                certificate_types[0],
+                certificate_key=certificate_key,
+                certificate_key_password=certificate_key_password,
+                should_print=False,
+            )
+            certificates.append(certificate)
+        return certificates
+
+    def _get_or_create_profiles(
```

Comment:
Used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L973-L1027